https://github.com/AndreRH/wine/commit/18da6525f5b92fbe4d3d27b6d3c1d2702fa0e40e

--- a/dlls/fexcore/unixlib.c
+++ b/dlls/fexcore/unixlib.c
@@ -101,10 +101,11 @@
     return 0;
 }
 
-static void invalidate_code_range ( void *args )
+static NTSTATUS invalidate_code_range ( void *args )
 {
     const struct invalidate_code_range_params *params = args;
     phangover_fex_invalidate_code_range(params->base, params->length);
+    return STATUS_SUCCESS;
 }
 
 const unixlib_entry_t __wine_unix_call_funcs[] =
