--- a/base/trace_event/trace_log.cc
+++ b/base/trace_event/trace_log.cc
@@ -2187,8 +2187,8 @@
 #if BUILDFLAG(USE_PERFETTO_CLIENT_LIBRARY)
 void TraceLog::OnSetup(const perfetto::DataSourceBase::SetupArgs& args) {
   AutoLock lock(track_event_lock_);
-  track_event_sessions_.emplace_back(args.internal_instance_index, *args.config,
-                                     args.backend_type);
+  track_event_sessions_.push_back({args.internal_instance_index, *args.config,
+                                     args.backend_type});
 }
 
 void TraceLog::OnStart(const perfetto::DataSourceBase::StartArgs&) {
