From e5996654ed48b385bc7f842d84d8b2ba72d29be1 Mon Sep 17 00:00:00 2001
From: vanhauser-thc <vh@thc.org>
Date: Sun, 6 Feb 2022 11:59:08 +0100
Subject: [PATCH] fix return

---
 hydra-http-form.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hydra-http-form.c b/hydra-http-form.c
index 224bf8dc..2fc6d60c 100644
--- a/hydra-http-form.c
+++ b/hydra-http-form.c
@@ -259,7 +259,7 @@ int32_t add_header(ptr_header_node *ptr_head, char *header, char *value, char ty
   ptr_header_node existing_hdr, new_ptr;
 
   if (!header || !value || !strlen(header) || !strlen(value))
-    return;
+    return 0;
 
   // get to the last header
   for (cur_ptr = *ptr_head; cur_ptr && cur_ptr->next; cur_ptr = cur_ptr->next)
