diff -uNr thc-hydra/configure thc-hydra.mod/configure
--- thc-hydra/configure	2022-08-18 19:18:12.734379960 +0800
+++ thc-hydra.mod/configure	2022-08-18 19:28:23.974379727 +0800
@@ -149,24 +149,24 @@
 # On macOS /usr/include only exists if one has installed the Command Line Tools package.
 # If this is an Xcode-only system we need to look inside the SDK for headers.
 SDK_PATH=""
-if [ "$SYSS" = "Darwin" -a ! -d "/usr/include" ]; then
+if [ "$SYSS" = "Darwin" -a ! -d "@TERMUX_PREFIX@/include" ]; then
   SDK_PATH=`xcrun --show-sdk-path`
 fi
-LIBDIRS=`cat /etc/ld.so.conf /etc/ld.so.conf.d/* 2> /dev/null | grep -v '^#' | sort | uniq`
+LIBDIRS=`cat @TERMUX_PREFIX@/etc/ld.so.conf @TERMUX_PREFIX@/etc/ld.so.conf.d/* 2> /dev/null | grep -v '^#' | sort | uniq`
 if [ "$SIXFOUR" = "64" ]; then
-  LIBDIRS="$LIBDIRS /lib64 /usr/lib64 /usr/local/lib64 /opt/local/lib64"
+  LIBDIRS="$LIBDIRS @TERMUX_PREFIX@/lib64"
 fi
 if [ -d "/Library/Developer/CommandLineTools/usr/lib" ]; then
   LIBDIRS="$LIBDIRS /Library/Developer/CommandLineTools/usr/lib /Library/Developer/CommandLineTools/lib"
 fi
-LIBDIRS="$LIBDIRS /lib /usr/lib /usr/local/lib /opt/local/lib /mingw64/lib /mingw64/bin"
-INCDIRS="$SDK_PATH/usr/include /usr/local/include /opt/include /opt/local/include /mingw64/include"
-if [ -n "$PREFIX" ]; then
-    if [ -d "$PREFIX/lib" ]; then
-        LIBDIRS="$LIBDIRS $PREFIX/lib"
+LIBDIRS="$LIBDIRS @TERMUX_PREFIX@/lib"
+INCDIRS="$SDK_PATH/usr/include @TERMUX_PREFIX@/include"
+if [ -n "$TERMUX_PREFIX" ]; then
+    if [ -d "$TERMUX_PREFIX/lib" ]; then
+        LIBDIRS="$LIBDIRS $TERMUX_PREFIX/lib"
     fi
-    if [ -d "$PREFIX/include" ]; then
-        INCDIRS="$INCDIRS $PREFIX/include"
+    if [ -d "$TERMUX_PREFIX/include" ]; then
+        INCDIRS="$INCDIRS $TERMUX_PREFIX/include"
     fi
 fi
 STRIP="strip"
@@ -197,7 +197,7 @@
 else
   for i in $LIBDIRS \
   /*ssl /usr/*ssl /opt/*ssl /usr/local/*ssl /opt/local/*ssl \
-  /*ssl/lib /usr/*ssl/lib /opt/*ssl/lib /usr/local/*ssl/lib /opt/local/*ssl/lib
+  /*ssl/lib /usr/*ssl/lib /opt/*ssl/lib /usr/local/*ssl/lib @TERMUX_PREFIX@/lib/*ssl
   do
     if [ "X" = "X$SSL_PATH" ]; then
         if [ -f "$i/libssl.so" -o -f "$i/libssl.dylib" -o -f "$i/libssl.a" ]; then
@@ -205,7 +205,7 @@
         fi
     fi
     if [ "X" = "X$SSL_PATH" ]; then
-        TMP_LIB=`/bin/ls $i/libssl.so* /bin/cygssl*.dll 2> /dev/null | grep ssl`
+        TMP_LIB=`@TERMUX_PREFIX@/bin/ls $i/libssl.so* @TERMUX_PREFIX@/bin/cygssl*.dll 2> /dev/null | grep ssl`
         if [ -n "$TMP_LIB" ]; then
           SSL_PATH="$i"
         fi
@@ -229,7 +229,7 @@
       SSL_PATH="$i"
       SSLNEW=`grep SHA256_CTX $i/openssl/sha.h 2> /dev/null`
 else
-  for i in $INCDIRS /*ssl/include /opt/*ssl/include /usr/*ssl/include /usr/local/*ssl/include
+  for i in $INCDIRS /*ssl/include /opt/*ssl/include /usr/*ssl/include @TERMUX_PREFIX@/include/*ssl
   do
     if [ "X" = "X$SSL_IPATH" ]; then
         if [ -f "$i/openssl/ssl.h" ]; then
@@ -447,7 +447,7 @@
   done
   POSTGRES_IPATH=
   for i in $INCDIRS \
-    /opt/p*sql*/include /usr/*p*sql*/include /usr/local/*psql*/include /mingw64/include
+    /opt/p*sql*/include /usr/*p*sql*/include /usr/local/*psql*/include @TERMUX_PREFIX@/include
   do
     if [ "X" = "X$POSTGRES_IPATH" ]; then
         if [ -f "$i/libpq-fe.h" ]; then
@@ -592,10 +592,10 @@
     SVN_IPATH=""
     APR_IPATH=""
 fi
-if [ "$SVN_IPATH" = "/usr/include" ]; then
+if [ "$SVN_IPATH" = "@TERMUX_PREFIX@/include" ]; then
     SVN_IPATH=""
 fi
-if [ "$APR_IPATH" = "/usr/include" ]; then
+if [ "$APR_IPATH" = "@TERMUX_PREFIX@/include" ]; then
     APR_IPATH=""
 fi

@@ -816,7 +816,7 @@
     echo "                                      ... NOT found, module sapr3 disabled"
     echo "Get it from http://www.sap.com/solutions/netweaver/linux/eval/index.asp"
 fi
-if [ "$SAPR3_IPATH" = "/usr/include" ]; then
+if [ "$SAPR3_IPATH" = "@TERMUX_PREFIX@/include" ]; then
     SAPR3_IPATH=""
 fi

@@ -864,7 +864,7 @@
     echo "                                      ... NOT found, module ssh disabled"
     echo 'Get it from http://www.libssh.org'
 fi
-if [ "$SSH_IPATH" = "/usr/include" ]; then
+if [ "$SSH_IPATH" = "@TERMUX_PREFIX@/include" ]; then
     SSH_IPATH=""
 fi

@@ -1304,10 +1304,10 @@


 if [ "X" = "X$XHYDRA_SUPPORT" ]; then
-    echo "Checking for GUI req's (pkg-config/gtk+-2.0) ..."
+    echo "Checking for GUI req's (pkg-config/gtk2) ..."
   XHYDRA_SUPPORT=`$PKG_CONFIG --help > /dev/null 2>&1 || echo disabled`
   if [ "X" = "X$XHYDRA_SUPPORT" ]; then
-     XHYDRA_SUPPORT=`$PKG_CONFIG --modversion gtk+-2.0 2> /dev/null`
+     XHYDRA_SUPPORT=`$PKG_CONFIG --modversion gtk2> /dev/null`
   else
      XHYDRA_SUPPORT=""
   fi
@@ -1711,12 +1711,12 @@
 if [ -n "$SMBC_PATH" ]; then
     XLIBS="$XLIBS -lsmbclient"
 fi
-if [ -d /usr/kerberos/include ]; then
+if [ -d @TERMUX_PREFIX@/include/kerberos/ ]; then
   XIPATHS="$XIPATHS -I/usr/kerberos/include"
 fi

 if [ "X" = "X$PREFIX" ]; then
-    PREFIX="/usr/local"
+    PREFIX="@TERMUX_PREFIX@"
 fi

 if [ "X" = "X$XHYDRA_SUPPORT" -o "Xdisable" = "X$XHYDRA_SUPPORT" ]; then
@@ -1726,7 +1726,7 @@
 fi

 if [ "X" != "X$FHS" ]; then
-  PREFIX=/usr
+  PREFIX=@TERMUX_PREFIX@
   echo "Hydra will be installed into .../bin of: $PREFIX"
   echo "  (according to the File System Hierarchy Standard)"
   echo
@@ -1747,13 +1747,13 @@
 fi

 if [ "$SYSS" = "SunOS" ]; then
-   XLIBS="$XLIBS -lrt"
+   XLIBS="$XLIBS"
 fi

 echo "Writing Makefile.in ..."
 if [ "X" != "X$FHS" ]; then
-  echo "MANDIR = /share/man/man1" >> Makefile.in
-  echo "DATADIR = /share/hydra" >> Makefile.in
+  echo "MANDIR = @TERMUX_PREFIX@/share/man/man1" >> Makefile.in
+  echo "DATADIR = @TERMUX_PREFIX@/share/hydra" >> Makefile.in
 fi
 echo "XDEFINES=$XDEFINES $MATH $SYS_PARAM" >> Makefile.in
 echo "XLIBS=$XLIBS" >> Makefile.in
