diff --git a/crates/llama-cpp-server/build.rs b/crates/llama-cpp-server/build.rs
index 35a4edb..3837363 100644
--- a/crates/llama-cpp-server/build.rs
+++ b/crates/llama-cpp-server/build.rs
@@ -12,6 +12,14 @@ fn main() {
     config.profile("Release");
     config.define("GGML_NATIVE", "OFF");
     config.define("GGML_NATIVE_DEFAULT", "OFF");
+    config.define("GGML_BLAS", "ON");
+    config.define("GGML_BLAS_VENDOR", "OpenBLAS");
+    config.define("GGML_OPENMP", "ON");
+    config.define("GGML_AVX", "OFF");
+    config.define("GGML_AVX2", "OFF");
+    config.define("GGML_AVX512", "OFF");
+    config.define("GGML_FMA", "OFF");
+    config.define("GGML_F16C", "OFF");
     config.define("BUILD_SHARED_LIBS", "OFF");
 
     if cfg!(target_os = "macos") {
