Only search packages from TUR repository when auto-updating.
--- a/scripts/bin/update-packages
+++ b/scripts/bin/update-packages
@@ -415,6 +417,11 @@ _should_update() {
 		termux_error_exit "directory '${pkg_dir}' is not a package."
 	fi
 
+	if [[ "$(basename "$(realpath "$(dirname "${pkg_dir}")")")" != *"tur"* ]]; then
+		echo "INFO: Skipping '$(basename "${pkg_dir}")', it is not a TUR package."
+		return 1
+	fi
+
 	# `[[ "$(<"$file")" =~ regex ]]` is bash builtin solution for `grep -q` without spawning extra process
 	# it is needed to pre-fetch and filter packages as fast as possible (it is 6.5 times faster than `grep`)
 	case "${PACKAGES_OPT_OUT:-}" in
@@ -471,7 +478,7 @@ if [[ "$1" == "@all" ]]; then
 	# pre-cache the latest tags and eliminate packages without updates
 	# when it is possible to determine that from the API response alone.
 	_fetch_and_cache_tags
-	for channel in $(jq --raw-output 'del(.pkg_format) | keys | .[]' "${TERMUX_SCRIPTDIR}/repo.json"); do
+	for channel in "tur" "tur-on-device" "tur-multilib" "tur-hacking"; do
 		for pkg_dir in "${channel}"/*; do
 			# Check if we should be putting this package into the list.
 			# Skip it if:
@@ -489,7 +496,7 @@ fi
 for pkg_dir in "$@"; do
 	_unix_millis="$(date +%10s%3N)"
 	if [[ ! -d "${pkg_dir}" ]]; then # If only package name is given, try to find it's directory.
-		for channel in $(jq --raw-output 'del(.pkg_format) | keys | .[]' "${TERMUX_SCRIPTDIR}/repo.json"); do
+		for channel in "tur" "tur-on-device" "tur-multilib" "tur-hacking"; do
 			if [[ -d "${channel}/${pkg_dir}" ]]; then
 				pkg_dir="${channel}/${pkg_dir}"
 				break
Sleep 1s after update in order to not reach GitHub rate limit.
--- a/scripts/bin/update-packages
+++ b/scripts/bin/update-packages
@@ -130,6 +130,7 @@ _update() {
 	exec {IPC_FIFO_FD}<>"$IPC_FIFO"
 	termux_pkg_auto_update 3>& "${IPC_FIFO_FD}"
 	exec {IPC_FIFO_FD}>&-
+	sleep 1s
 }
 
 declare -A _LATEST_TAGS=()
Do not run check-updated step, which will make it easy to reach GitHub rate limit.
--- a/scripts/bin/update-packages
+++ b/scripts/bin/update-packages
@@ -316,6 +317,7 @@ _fetch_and_cache_tags() {
 }
 
 _check_updated() {
+	return 1
 	local latest_tag="${_LATEST_TAGS[${1##*/}]:-}"
 
 	if [[ -z "${latest_tag}" ]]; then
--- a/scripts/bin/build-package-dry-run-simulation.sh
+++ b/scripts/bin/build-package-dry-run-simulation.sh
@@ -12,6 +12,10 @@ TERMUX_PACKAGES_DIRECTORIES="
 packages
 root-packages
 x11-packages
+tur
+tur-on-device
+tur-multilib
+tur-hacking
 "
 
 # Please keep synchronized with the logic of lines 468-547 of 'build-package.sh'.
