Only search packages from TUR repository when auto-updating.
--- a/scripts/bin/update-packages
+++ b/scripts/bin/update-packages
@@ -307,7 +307,7 @@
 
 if [[ "$1" == "@all" ]]; then
 	_fetch_and_cache_tags
-	for repo_dir in $(jq --raw-output 'del(.pkg_format) | keys | .[]' "${TERMUX_SCRIPTDIR}/repo.json"); do
+	for repo_dir in "tur" "tur-on-device" "tur-multilib" "tur-hacking"; do
 		for pkg_dir in "${repo_dir}"/*; do
 			_unix_millis=$(date +%s%N | cut -b1-13)
 			! _should_update "${pkg_dir}" && continue # Skip if not needed.
@@ -327,7 +327,7 @@
 else
 	for pkg in "$@"; do
 		if [ ! -d "${pkg}" ]; then # If only package name is given, try to find it's directory.
-			for repo_dir in $(jq --raw-output 'del(.pkg_format) | keys | .[]' "${TERMUX_SCRIPTDIR}/repo.json"); do
+			for repo_dir in "tur" "tur-on-device" "tur-multilib" "tur-hacking"; do
 				if [ -d "${repo_dir}/${pkg}" ]; then
 					pkg="${repo_dir}/${pkg}"
 					break
@@ -352,7 +352,7 @@
 	local body
 
 	if [[ "${assignee:-termuxbot2}" == "termuxbot2" ]]; then
-		assignee="MrAdityaAlok" # Assign myself if termuxbot2 is the actor.
+		assignee="licy183" # Assign myself if termuxbot2 is the actor.
 	fi
 
 	body="$(
Sleep 1s after update in order to not reach GitHub rate limit.
--- a/scripts/bin/update-packages
+++ b/scripts/bin/update-packages
@@ -118,6 +118,7 @@
 	echo # Newline.
 	echo "INFO: Updating ${TERMUX_PKG_NAME} [Current version: ${TERMUX_PKG_VERSION}]"
 	termux_pkg_auto_update
+	sleep 1s
 }
 
 declare -A _LATEST_TAGS=()
Do not run check-updated step, which will make it easy to reach GitHub rate limit.
--- a/scripts/bin/update-packages
+++ b/scripts/bin/update-packages
@@ -195,6 +195,7 @@
 }
 
 _check_updated() {
+	return 1
 	if [[ -n "${_LATEST_TAGS[${1##*/}]:-}" ]]; then
 		(
 			set +eu
