Add some dependencies required by TUR-on-Device.
--- a/scripts/setup-termux.sh
+++ b/scripts/setup-termux.sh
@@ -43,6 +43,19 @@
 PACKAGES+=" valac"
 PACKAGES+=" xmlto"                      # Needed by git's manpage generation
 
+# Tier 3: requirements for community repos.
+PACKAGES+=" tur-repo"
+PACKAGES+=" x11-repo"
+echo "" >> /system/etc/static-dns-hosts.txt
+echo "# Community TUR Repo" >> /system/etc/static-dns-hosts.txt
+echo "tur.kcubeterm.com" >> /system/etc/static-dns-hosts.txt
+echo "turdl.kcubeterm.com" >> /system/etc/static-dns-hosts.txt
+echo "termux-user-repository.github.io"  >> /system/etc/static-dns-hosts.txt
+
+update-static-dns
+sed -i "s/packages.termux.dev/packages-cf.termux.dev/g" /data/data/com.termux/files/usr/etc/apt/sources.list
+apt update
+apt dist-upgrade -o Dpkg::Options::=--force-confnew -y
+sed -i "s/packages.termux.dev/packages-cf.termux.dev/g" /data/data/com.termux/files/usr/etc/apt/sources.list
 apt update
-apt dist-upgrade -y
 apt install -y $PACKAGES
