--- a/scripts/build/setup/termux_setup_python_pip.sh
+++ b/scripts/build/setup/termux_setup_python_pip.sh
@@ -53,8 +53,8 @@
 
 		if [ ! -d "$TERMUX_PYTHON_CROSSENV_PREFIX" ]; then
 			cd "$TERMUX_PYTHON_CROSSENV_SRCDIR"
-			/usr/bin/python${TERMUX_PYTHON_VERSION} -m crossenv \
-                		"$TERMUX_PREFIX/bin/python${TERMUX_PYTHON_VERSION}" \
+			python${TERMUX_PYTHON_VERSION} -m crossenv \
+                "$TERMUX_PREFIX/bin/python${TERMUX_PYTHON_VERSION}" \
 				"${TERMUX_PYTHON_CROSSENV_PREFIX}"
 		fi
 		. "${TERMUX_PYTHON_CROSSENV_PREFIX}/bin/activate"
