--- a/scripts/run-docker.sh
+++ b/scripts/run-docker.sh
@@ -96,7 +96,7 @@ if [ "$UNAME" = Darwin ]; then
 	SEC_OPT=""
 else
 	REPOROOT="$(dirname $(readlink -f $0))/../"
-	SEC_OPT=" --security-opt seccomp=$REPOROOT/scripts/profile.json --security-opt apparmor=_custom-termux-package-builder-$CONTAINER_NAME --cap-add CAP_SYS_ADMIN --device /dev/fuse"
+	SEC_OPT=" --privileged --security-opt seccomp=unconfined --cap-add CAP_SYS_ADMIN --device /dev/fuse"
 fi
 
 if [ "${CI:-}" = "true" ]; then
@@ -135,6 +135,7 @@ APPARMOR_PARSER=""
 if command -v apparmor_parser > /dev/null; then
 	APPARMOR_PARSER="apparmor_parser"
 fi
+APPARMOR_PARSER=""
 
 if [ -z "$APPARMOR_PARSER" ] || ! $SUDO aa-status --enabled; then
 	echo "WARNING: apparmor_parser not found, AppArmor profiles will not be loaded!"
