Description: Fix the build on GNU/kFreeBSD.
Forwarded: no
Author: Petr Salinger <Petr.Salinger@seznam.cz>
Last-Update: 2007-02-20

--- freealut-1.1.0.orig/configure.ac
+++ freealut-1.1.0/configure.ac
@@ -82,7 +82,7 @@
 AC_DEFINE([ALUT_BUILD_LIBRARY], [1], [Define to 1 if you want to build the ALUT DLL.])
 
 # Checks for libraries. (not perfect yet)
-AC_SEARCH_LIBS([pthread_self], [pthread])
+AC_SEARCH_LIBS([pthread_create], [pthread])
 AC_SEARCH_LIBS([alGetError], [openal32 openal])
 
 ################################################################################
