diff --git a/Makefile b/Makefile
index 9cbedc2..7fba2c2 100644
--- a/Makefile
+++ b/Makefile
@@ -6,7 +6,7 @@ CFLAGS   += -std=c99 -pedantic -Wall -Wextra -DJSMN_STRICT
 LDFLAGS  ?=
 LDLIBS    = $(LDFLAGS) -lm -lxcb -lxcb-util -lxcb-keysyms -lxcb-icccm -lxcb-ewmh -lxcb-randr -lxcb-xinerama -lxcb-shape
 
-PREFIX    ?= /usr/local
+PREFIX    ?= /data/data/com.termux/files/usr
 BINPREFIX ?= $(PREFIX)/bin
 MANPREFIX ?= $(PREFIX)/share/man
 DOCPREFIX ?= $(PREFIX)/share/doc/bspwm
diff --git a/src/bspwm.h b/src/bspwm.h
index 9ad0475..bba9e2e 100644
--- a/src/bspwm.h
+++ b/src/bspwm.h
@@ -32,7 +32,7 @@
 #define CONFIG_HOME_ENV          "XDG_CONFIG_HOME"
 #define RUNTIME_DIR_ENV          "XDG_RUNTIME_DIR"
 
-#define STATE_PATH_TPL           "/tmp/bspwm%s_%i_%i-state"
+#define STATE_PATH_TPL           "/data/data/com.termux/files/usr/tmp/bspwm%s_%i_%i-state"
 
 #define ROOT_EVENT_MASK     (XCB_EVENT_MASK_SUBSTRUCTURE_REDIRECT | XCB_EVENT_MASK_SUBSTRUCTURE_NOTIFY | XCB_EVENT_MASK_STRUCTURE_NOTIFY | XCB_EVENT_MASK_BUTTON_PRESS)
 #define CLIENT_EVENT_MASK   (XCB_EVENT_MASK_PROPERTY_CHANGE | XCB_EVENT_MASK_FOCUS_CHANGE)
diff --git a/src/common.h b/src/common.h
index f44eef8..6992473 100644
--- a/src/common.h
+++ b/src/common.h
@@ -25,7 +25,7 @@
 #ifndef BSPWM_COMMON_H
 #define BSPWM_COMMON_H
 
-#define SOCKET_PATH_TPL  "/tmp/bspwm%s_%i_%i-socket"
+#define SOCKET_PATH_TPL  "/data/data/com.termux/files/usr/tmp/bspwm%s_%i_%i-socket"
 #define SOCKET_ENV_VAR   "BSPWM_SOCKET"
 
 #define FAILURE_MESSAGE  "\x07"
