--- a/src/paccache.sh.in
+++ b/src/paccache.sh.in
@@ -134,16 +134,7 @@
 m4_include(../lib/size_to_human.sh)
 
 runasroot() {
-	if (( needsroot && EUID != 0 )); then
-		if ! type -p sudo; then
-			error 'Cannot find the sudo binary!'
-			die 'Either install "sudo" or run as root.'
-		else
-			sudo "$@"
-		fi
-	else
 		"$@"
-	fi
 }
 
 summarize() {
--- a/src/pacscripts.sh.in
+++ b/src/pacscripts.sh.in
@@ -63,16 +63,7 @@
 }
 
 runasroot() {
-	if (( EUID != 0 )); then
-		if ! type -p sudo; then
-			error 'Cannot find the sudo binary!'
-			die "${myname} requires root privileges. Either install \"sudo\" or run as root."
-		else
-			sudo "$@"
-		fi
-	else
 		"$@"
-	fi
 }
 
 print_db() {
