This reverts commit 25db1c5d49dc339e9b5a25514c198a524bd05484.

--- a/dlls/ntdll/unix/virtual.c
+++ b/dlls/ntdll/unix/virtual.c
@@ -2857,8 +2857,6 @@
     void *base = wine_server_get_ptr( image_info->base );
     NTSTATUS status;
     ULONG_PTR start, end;
-    BOOL top_down = (image_info->image_charact & IMAGE_FILE_DLL) &&
-                    (image_info->image_flags & IMAGE_FLAGS_ImageDynamicallyRelocated);
 
     limit_low = max( limit_low, (ULONG_PTR)address_space_start );  /* make sure the DOS area remains free */
     if (!limit_high) limit_high = (ULONG_PTR)user_space_limit;
@@ -2867,7 +2865,6 @@
 
     if (base && (ULONG_PTR)base == image_info->base)
     {
-        if (top_down) alloc_type |= MEM_TOP_DOWN;
         status = map_view( view_ret, base, size, alloc_type, vprot, limit_low, limit_high, 0 );
         if (!status) return status;
     }
@@ -2886,13 +2883,13 @@
     }
     if (start < end && (start != limit_low || end != limit_high))
     {
-        status = map_view( view_ret, NULL, size, top_down ? MEM_TOP_DOWN : 0, vprot, start, end, 0 );
+        status = map_view( view_ret, NULL, size, alloc_type, vprot, start, end, 0 );
         if (!status) return status;
     }
 
     /* then any suitable address */
 
-    return map_view( view_ret, NULL, size, top_down ? MEM_TOP_DOWN : 0, vprot, limit_low, limit_high, 0 );
+    return map_view( view_ret, NULL, size, alloc_type, vprot, limit_low, limit_high, 0 );
 }
 
 
