--- a/content/browser/geolocation/geolocation_service_impl.cc
+++ b/content/browser/geolocation/geolocation_service_impl.cc
@@ -227,7 +227,7 @@
   bool has_precise_permission =
       permission_level == GeolocationPermissionLevel::kPrecise;
 
-  if (base::FeatureList::IsEnabled(features::kGeolocationProxy)) {
+  if (base::FeatureList::IsEnabled(::features::kGeolocationProxy)) {
     mojo::PendingRemote<device::mojom::Geolocation> geolocation_impl_remote;
     geolocation_context->BindGeolocation(
         geolocation_impl_remote.InitWithNewPipeAndPassReceiver(),
@@ -285,7 +285,7 @@
         ->UnsubscribeFromPermissionResultChange(subscription_id_);
     // When kGeolocationProxy is enabled, DecrementActivityCount is managed
     // by the GeolocationProxy lifecycle.
-    if (!base::FeatureList::IsEnabled(features::kGeolocationProxy)) {
+    if (!base::FeatureList::IsEnabled(::features::kGeolocationProxy)) {
       DecrementActivityCount();
     }
   }
@@ -296,7 +296,7 @@
 void GeolocationServiceImpl::OnDisconnected() {
   // When kGeolocationProxy is enabled, we do not need to perform any cleanup
   // here because active connection lifetimes are managed by the proxies.
-  if (!base::FeatureList::IsEnabled(features::kGeolocationProxy) &&
+  if (!base::FeatureList::IsEnabled(::features::kGeolocationProxy) &&
       receiver_set_.empty()) {
     DecrementActivityCount();
   }
--- a/content/browser/generic_sensor/frame_sensor_provider_proxy.cc
+++ b/content/browser/generic_sensor/frame_sensor_provider_proxy.cc
@@ -117,7 +117,7 @@
   }
 
   if (base::FeatureList::IsEnabled(
-          features::kSeverSensorConnectionsOnPermissionRevocation) &&
+          ::features::kSeverSensorConnectionsOnPermissionRevocation) &&
       !permission_subscription_id_) {
     permission_subscription_id_ =
         permission_controller->SubscribeToPermissionResultChange(
@@ -133,7 +133,7 @@
 
   mojo::PendingRemote<device::mojom::SensorConnectionWatcher> watcher;
   if (base::FeatureList::IsEnabled(
-          features::kSeverSensorConnectionsOnPermissionRevocation)) {
+          ::features::kSeverSensorConnectionsOnPermissionRevocation)) {
     watcher_receivers_.Add(this, watcher.InitWithNewPipeAndPassReceiver());
   }
 
