diff -ruN octave-7.3.0.orig/liboctave/system/oct-env.cc octave-7.3.0/liboctave/system/oct-env.cc
--- octave-7.3.0.orig/liboctave/system/oct-env.cc	2022-11-03 03:19:56.000000000 +0900
+++ octave-7.3.0/liboctave/system/oct-env.cc	2022-11-18 18:03:37.000000000 +0900
@@ -235,7 +235,7 @@
         tempd = P_tmpdir;
 #else
       if (tempd.empty ())
-        tempd = "/tmp";
+        tempd = "@TERMUX_PREFIX@/tmp";
 #endif
 
 #endif
