diff -ruN octave-7.3.0.orig/liboctave/system/oct-password.cc octave-7.3.0/liboctave/system/oct-password.cc
--- octave-7.3.0.orig/liboctave/system/oct-password.cc	2022-11-03 03:19:56.000000000 +0900
+++ octave-7.3.0/liboctave/system/oct-password.cc	2022-11-18 18:03:37.000000000 +0900
@@ -31,6 +31,15 @@
 
 #if defined (HAVE_PWD_H)
 #  include <pwd.h>
+#  ifdef getpwnam
+#  undef getpwnam
+#  endif
+#  ifdef getpwuid
+#  undef getpwuid
+#  endif
+#  ifdef getpwuid_r
+#  undef getpwuid_r
+#  endif
 #endif
 
 #include "lo-error.h"
