--- ./deps/LZMA-SDK/C/7zCrc.c
+++ ./deps/LZMA-SDK/C/7zCrc.c
@@ -81,8 +81,9 @@
         #define USE_ARM64_CRC
     #endif
     #endif
-  #elif (defined(__clang__) && (__clang_major__ >= 3)) \
-     || (defined(__GNUC__) && (__GNUC__ > 4))
+  #elif ( (defined(__clang__) && (__clang_major__ >= 3)) ||	 \
+          (defined(__GNUC__)  && (__GNUC__ > 4)        )    ) && \
+        (__ARM_ARCH >= 8)
       #if !defined(__ARM_FEATURE_CRC32)
         #define __ARM_FEATURE_CRC32 1
           #if (!defined(__clang__) || (__clang_major__ > 3)) // fix these numbers
