--- a/ddclient.in
+++ b/ddclient.in
@@ -1195,7 +1195,7 @@
         'examples'   => \&nic_nsupdate_examples,
         'cfgvars' => {
             %{$cfgvars{'protocol-common-defaults'}},
-            'login' => setv(T_LOGIN,  0, '/usr/bin/nsupdate', undef),
+            'login' => setv(T_LOGIN,  0, '@TERMUX_PREFIX@/bin/nsupdate', undef),
             'tcp'   => setv(T_BOOL,   0, 0,                   undef),
             'ttl'   => setv(T_NUMBER, 0, 600,                 undef),
             'zone'  => setv(T_STRING, 1, undef,               undef),
@@ -2777,7 +2777,7 @@
 
     # Hard code to /tmp rather than use system TMPDIR to protect from malicious
     # shell instructions in TMPDIR environment variable.  All systems should have /tmp.
-    $tfh = File::Temp->new(DIR => '/tmp',
+    $tfh = File::Temp->new(DIR => '@TERMUX_PREFIX@/tmp',
                            TEMPLATE => 'ddclient_XXXXXXXXXX');
     $tmpfile = $tfh->filename;
 
