--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -1543,20 +1543,6 @@
         # TODO(thakis): Only for no_chromium_code? http://crbug.com/912662
         "-Wno-ignored-pragma-optimize",
       ]
-
-      if (!is_nacl) {
-        cflags += [
-          # TODO(crbug.com/1343975) Evaluate and possibly enable.
-          "-Wno-deprecated-builtins",
-
-          # TODO(crbug.com/1352183) Evaluate and possibly enable.
-          "-Wno-bitfield-constant-conversion",
-        ]
-        if (is_win) {
-          # TODO(https://crbug.com/1363759): Evaluate and possibly enable.
-          cflags += [ "-Wno-deprecated-copy-with-user-provided-copy" ]
-        }
-      }
     }
   }
 }
