https://github.com/blender/blender/commit/eb9209c1f5701844eaf1e0652e48a6f7dd1594a9

diff --git a/intern/ffmpeg/ffmpeg_compat.h b/intern/ffmpeg/ffmpeg_compat.h
index 53f28239760d..53c4e100e260 100644
--- a/intern/ffmpeg/ffmpeg_compat.h
+++ b/intern/ffmpeg/ffmpeg_compat.h
@@ -123,6 +123,17 @@ int64_t av_get_pts_from_frame(AVFrame *picture)
   return timestamp_from_pts_or_dts(picture->pts, picture->pkt_dts);
 }
 
+/*  Duration of the frame, in the same units as pts. 0 if unknown. */
+FFMPEG_INLINE
+int64_t av_get_frame_duration_in_pts_units(const AVFrame *picture)
+{
+#if LIBAVUTIL_VERSION_INT < AV_VERSION_INT(57, 30, 100)
+  return picture->pkt_duration;
+#else
+  return picture->duration;
+#endif
+}
+
 /* -------------------------------------------------------------------- */
 /** \name Deinterlace code block
  *
diff --git a/source/blender/imbuf/intern/anim_movie.cc b/source/blender/imbuf/intern/anim_movie.cc
index c10f71c177ed..0e28a0ece50c 100644
--- a/source/blender/imbuf/intern/anim_movie.cc
+++ b/source/blender/imbuf/intern/anim_movie.cc
@@ -956,7 +956,7 @@ static AVFrame *ffmpeg_frame_by_pts_get(struct anim *anim, int64_t pts_to_search
 
   const bool backup_frame_ready = anim->pFrame_backup_complete;
   const int64_t recent_start = av_get_pts_from_frame(anim->pFrame);
-  const int64_t recent_end = recent_start + anim->pFrame->pkt_duration;
+  const int64_t recent_end = recent_start + av_get_frame_duration_in_pts_units(anim->pFrame);
   const int64_t backup_start = backup_frame_ready ? av_get_pts_from_frame(anim->pFrame_backup) : 0;
 
   AVFrame *best_frame = nullptr;
@@ -1165,7 +1165,7 @@ static bool ffmpeg_is_first_frame_decode(struct anim *anim)
 static void ffmpeg_scan_log(struct anim *anim, int64_t pts_to_search)
 {
   int64_t frame_pts_start = av_get_pts_from_frame(anim->pFrame);
-  int64_t frame_pts_end = frame_pts_start + anim->pFrame->pkt_duration;
+  int64_t frame_pts_end = frame_pts_start + av_get_frame_duration_in_pts_units(anim->pFrame);
   av_log(anim->pFormatCtx,
          AV_LOG_DEBUG,
          "  SCAN WHILE: PTS range %" PRId64 " - %" PRId64 " in search of %" PRId64 "\n",
