--- a/net/dns/host_resolver_cache.h
+++ b/net/dns/host_resolver_cache.h
@@ -143,13 +143,15 @@
     }
 
     bool operator()(const Key& lhs, const KeyRef& rhs) const {
-      return std::tie(lhs.domain_name, lhs.network_anonymization_key) <
+      const std::string_view lhs_domain_name = lhs.domain_name;
+      return std::tie(lhs_domain_name, lhs.network_anonymization_key) <
              std::tie(rhs.domain_name, *rhs.network_anonymization_key);
     }
 
     bool operator()(const KeyRef& lhs, const Key& rhs) const {
+      const std::string_view rhs_domain_name = rhs.domain_name;
       return std::tie(lhs.domain_name, *lhs.network_anonymization_key) <
-             std::tie(rhs.domain_name, rhs.network_anonymization_key);
+             std::tie(rhs_domain_name, rhs.network_anonymization_key);
     }
   };
 
