--- a/src/mongo/shell/mongo_main.cpp
+++ b/src/mongo/shell/mongo_main.cpp
@@ -586,7 +586,7 @@
         GetTempPathA(sizeof tempFolder, tempFolder);
         sb << tempFolder << "mongo_edit" << time(0) + i << ".js";
 #else
-        sb << "/tmp/mongo_edit" << time(nullptr) + i << ".js";
+        sb << "@TERMUX_PREFIX@/tmp/mongo_edit" << time(nullptr) + i << ".js";
 #endif
         filename = sb.str();
         if (!::mongo::shell_utils::fileExists(filename))
--- a/src/mongo/db/server_options.h
+++ b/src/mongo/db/server_options.h
@@ -84,7 +84,7 @@
 
     bool noUnixSocket = false;    // --nounixsocket
     bool doFork = false;          // --fork
-    std::string socket = "/tmp";  // UNIX domain socket directory
+    std::string socket = "@TERMUX_PREFIX@/tmp";  // UNIX domain socket directory
     std::string transportLayer;   // --transportLayer (must be either "asio" or "legacy")
 
     size_t maxConns = DEFAULT_MAX_CONN;  // Maximum number of simultaneous open connections.
--- a/src/mongo/db/storage/storage_options.cpp
+++ b/src/mongo/db/storage/storage_options.cpp
@@ -81,8 +81,8 @@
 const char* StorageGlobalParams::kDefaultDbPath = "\\data\\db\\";
 const char* StorageGlobalParams::kDefaultConfigDbPath = "\\data\\configdb\\";
 #else
-const char* StorageGlobalParams::kDefaultDbPath = "/data/db";
-const char* StorageGlobalParams::kDefaultConfigDbPath = "/data/configdb";
+const char* StorageGlobalParams::kDefaultDbPath = "@TERMUX_PREFIX@/var/lib/mongodb/db";
+const char* StorageGlobalParams::kDefaultConfigDbPath = "@TERMUX_PREFIX@/var/lib/mongodb/configdb";
 #endif
 
 }  // namespace mongo
