--- a/flang/CMakeLists.txt
+++ b/flang/CMakeLists.txt
@@ -142,8 +142,8 @@
 
   get_errc_messages(LLVM_LIT_ERRC_MESSAGES)
 
-#Handle unittests when out-of-tree
-#LLVM_BUILD_MAIN_SRC_DIR - Path to llvm source when out-of-tree.
+ #Handle unittests when out-of-tree
+ #LLVM_BUILD_MAIN_SRC_DIR - Path to llvm source when out-of-tree.
   set(FLANG_GTEST_AVAIL 0)
   if (FLANG_INCLUDE_TESTS)
     set(UNITTEST_DIR ${LLVM_BUILD_MAIN_SRC_DIR}/utils/unittest)
@@ -199,7 +199,11 @@
   set(MLIR_MAIN_SRC_DIR ${LLVM_MAIN_SRC_DIR}/../mlir/include ) # --src-root
   set(MLIR_INCLUDE_DIR ${LLVM_MAIN_SRC_DIR}/../mlir/include ) # --includedir
   set(MLIR_TABLEGEN_OUTPUT_DIR ${CMAKE_BINARY_DIR}/tools/mlir/include)
-  set(MLIR_TABLEGEN_EXE $<TARGET_FILE:mlir-tblgen>)
+  if(MLIR_TABLEGEN)
+    set(MLIR_TABLEGEN_EXE ${MLIR_TABLEGEN})
+  else()
+    set(MLIR_TABLEGEN_EXE $<TARGET_FILE:mlir-tblgen>)
+  endif()
   include_directories(SYSTEM ${MLIR_INCLUDE_DIR})
   include_directories(SYSTEM ${MLIR_TABLEGEN_OUTPUT_DIR})
 endif()
