--- a/meson.build
+++ b/meson.build
@@ -4,7 +4,7 @@ project(
     'c',
     'cpp',
     'cython',
-    version: run_command(['generate_version.py', '--print'], check: true).stdout().strip(),
+    version: run_command(['@PYTHON@', 'generate_version.py', '--print'], check: true).stdout().strip(),
     license: 'BSD-3',
     meson_version: '>=1.2.1',
     default_options: ['buildtype=release', 'c_std=c11', 'warning_level=2'],
