diff -uNr flang-src/clang/lib/Driver/ToolChain.cpp flang-src.mod/clang/lib/Driver/ToolChain.cpp
--- flang-src/clang/lib/Driver/ToolChain.cpp
+++ flang-src.mod/clang/lib/Driver/ToolChain.cpp
@@ -632,7 +632,7 @@
 
   switch (Type) {
   case ToolChain::CST_Libcxx:
-    CmdArgs.push_back("-lc++");
+    CmdArgs.push_back("-lc++_shared");
     if (Args.hasArg(options::OPT_fexperimental_library))
       CmdArgs.push_back("-lc++experimental");
     break;
@@ -942,6 +942,7 @@

   if (staticFlangLibs)
     CmdArgs.push_back("-Bstatic");
+  CmdArgs.push_back("-landroid-complex-math");
   CmdArgs.push_back("-lflang");
   CmdArgs.push_back("-lflangrti");
   CmdArgs.push_back("-lpgmath");
