diff -uNr flang-src/llvm/tools/llvm-shlib/CMakeLists.txt flang-src.mod/llvm/tools/llvm-shlib/CMakeLists.txt
--- flang-src/llvm/tools/llvm-shlib/CMakeLists.txt
+++ flang-src.mod/llvm/tools/llvm-shlib/CMakeLists.txt
@@ -39,6 +39,7 @@
      OR ("${CMAKE_SYSTEM_NAME}" STREQUAL "OpenBSD")
      OR ("${CMAKE_SYSTEM_NAME}" STREQUAL "Fuchsia")
      OR ("${CMAKE_SYSTEM_NAME}" STREQUAL "DragonFly")
+     OR ("${CMAKE_SYSTEM_NAME}" STREQUAL "Android")
      OR ("${CMAKE_SYSTEM_NAME}" STREQUAL "SunOS")) # FIXME: It should be "GNU ld for elf"
     configure_file(
     ${CMAKE_CURRENT_SOURCE_DIR}/simple_version_script.map.in
