--- OpenImageIO-2.5.6.0/src/libOpenImageIO/imagebufalgo_draw.cpp	2023-12-01 11:16:11.000000000 +0530
+++ OpenImageIO-2.5.6.0.mod/src/libOpenImageIO/imagebufalgo_draw.cpp	2023-12-24 11:52:40.708918320 +0530
@@ -735,7 +735,7 @@
     = { "OPENIMAGEIO_FONTS", "HOME", "SystemRoot", "OpenImageIO_ROOT" };
 static const char* font_dir_prefixes[]
     = { "/Library/Fonts", "/System/Library/Fonts",
-        "C:/Windows",     "/usr",
+        "C:/Windows",     "@TERMUX_PREFIX@",
         "/usr/local",     "/opt/local" };
 static const char* font_dir_suffixes[]
     = { "fonts",       "Fonts",       "Library/Fonts",
