--- a/src/core.h
+++ b/src/core.h
@@ -40,9 +40,9 @@
 #endif
 
 // Define CPU
-#if defined( __x86_64__ ) || defined( _M_X64 )
+#if defined( __x86_64__ ) || defined( _M_X64 ) || defined( __i386__ ) || defined( _M_IX86 )
 	#define B2_CPU_X64
-#elif defined( __aarch64__ ) || defined( _M_ARM64 )
+#elif defined( __aarch64__ ) || defined( _M_ARM64 ) || defined( __arm__ ) || defined( _M_ARM )
 	#define B2_CPU_ARM
 #elif defined( __EMSCRIPTEN__ )
 	#define B2_CPU_WASM
@@ -61,6 +61,8 @@
 
 #if defined( B2_COMPILER_MSVC )
 	#define B2_BREAKPOINT __debugbreak()
+#elif defined( B2_PLATFORM_ANDROID )
+	#define B2_BREAKPOINT __builtin_trap()
 #elif defined( B2_PLATFORM_WASM )
 	#define B2_BREAKPOINT                                                                                                        \
 		do                                                                                                                       \
