--- node-v18.10.0/test/common/index.js	2022-09-28 20:36:06.000000000 +0530
+++ node-v18.10.0.mod/test/common/index.js	2022-10-06 10:31:06.867469751 +0530
@@ -126,6 +126,13 @@
 const isASan = process.config.variables.asan === 1;
 const isRiscv64 = process.arch === 'riscv64';
 const isDebug = process.features.debug;
+// The regex inside the function causes test-startup-empty-regexp-statics.js
+// to break. Probably not something Termux specific, but don't know why anyone
+// else hasn't complaint upstream
+function isPi() {
+  return false;
+}
+/*
 function isPi() {
   try {
     // Normal Raspberry Pi detection is to find the `Raspberry Pi` string in
@@ -139,6 +146,7 @@
     return false;
   }
 }
+*/
 
 // When using high concurrency or in the CI we need much more time for each connection attempt
 net.setDefaultAutoSelectFamilyAttemptTimeout(platformTimeout(net.getDefaultAutoSelectFamilyAttemptTimeout() * 10));
