--- a/source/mocks/dlfcn.h	2026-02-28 19:36:41.034241644 +0100
+++ b/source/mocks/dlfcn.h	2026-02-28 19:36:41.250240013 +0100
@@ -16,7 +16,14 @@
  */
 
 #include <dlfcn.h>
+#if defined(CONFIG_ENABLE_MOCKS) && (CONFIG_ENABLE_MOCKS == 0)
+#ifndef CXXTEST_MOCK_GLOBAL
+#define CXXTEST_MOCK_GLOBAL(ret, func, args, params) \
+  namespace T { inline ret func args { return ::func params; } }
+#endif
+#else
 #include <cxxtest/Mock.h>
+#endif
 CXXTEST_MOCK_GLOBAL(
 	int, dladdr,
 	(void *addr, Dl_info *info),
--- a/source/mocks/unistd.h	2026-02-28 19:36:41.058241462 +0100
+++ b/source/mocks/unistd.h	2026-02-28 19:36:41.290885246 +0100
@@ -16,7 +16,14 @@
  */
 
 #include <unistd.h>
+#if defined(CONFIG_ENABLE_MOCKS) && (CONFIG_ENABLE_MOCKS == 0)
+#ifndef CXXTEST_MOCK_GLOBAL
+#define CXXTEST_MOCK_GLOBAL(ret, func, args, params) \
+  namespace T { inline ret func args { return ::func params; } }
+#endif
+#else
 #include <cxxtest/Mock.h>
+#endif
 CXXTEST_MOCK_GLOBAL(
 	char *, getcwd,
 	(char *buf, size_t size),
