--- a/src/aptitude.h	2020-05-21 12:32:38.000000000 +0900
+++ b/src/aptitude.h	2023-03-23 18:32:58.208009137 +0900
@@ -29,6 +29,8 @@
 #endif
 
 # include <string.h>
+# include <gettext.h>
+# include <stdlib.h>
 
 #if ENABLE_NLS
 # include <libintl.h>
--- a/src/generic/apt/matching/parse.h	2020-05-21 12:32:38.000000000 +0900
+++ b/src/generic/apt/matching/parse.h	2023-03-23 18:32:58.204009137 +0900
@@ -22,6 +22,7 @@
 #define PARSE_H
 
 #include <cwidget/generic/util/ref_ptr.h>
+#include <vector>
 
 /** \file parse.h */
 
--- a/src/generic/views/download_progress.h
+++ b/src/generic/views/download_progress.h
@@ -26,6 +26,7 @@
 
 #include <sigc++/slot.h>
 
+#include <iostream>
 #include <optional>
 #include <string>
 #include <vector>
--- a/src/ui.cc	2020-05-21 12:32:38.000000000 +0900
+++ b/src/ui.cc	2023-03-23 18:32:58.256009139 +0900
@@ -694,8 +694,8 @@
 	{
 	  really_quit_active = true;
 	  prompt_yesno(_("Really quit Aptitude?"), false,
-		       cw::util::arg(sigc::bind(ptr_fun(do_really_quit_answer), true)),
-		       cw::util::arg(sigc::bind(ptr_fun(do_really_quit_answer), false)));
+		       cw::util::arg(sigc::bind(sigc::ptr_fun(do_really_quit_answer), true)),
+		       cw::util::arg(sigc::bind(sigc::ptr_fun(do_really_quit_answer), false)));
 	}
     }
   else
