diff -uNr a/src/Makefile.am b/src/Makefile.am
--- a/src/Makefile.am	2020-05-21 12:32:38.000000000 +0900
+++ b/src/Makefile.am	2023-03-23 19:33:12.832145507 +0900
@@ -14,7 +14,7 @@
 
 localedir = $(datadir)/locale
 AM_CPPFLAGS = -I$(top_builddir) -I$(srcdir) -I$(top_srcdir) -I$(top_srcdir)/src
-FILESYSTEM_LDFLAGS = -lstdc++fs
+FILESYSTEM_LDFLAGS = 
 DEFS = -DLOCALEDIR=\"$(localedir)\" @DEFS@
 
 bin_PROGRAMS=aptitude
