--- a/src/common/darktable.c
+++ b/src/common/darktable.c
@@ -1185,9 +1185,11 @@
           dt_print(DT_DEBUG_ALWAYS,
                    "[dt_init --threads] requested %d ompthreads restricted to %d",
             desired, possible);
+#ifdef _OPENMP
         dt_print(DT_DEBUG_ALWAYS,
                  "[dt_init --threads] using %d threads of %d for openmp parallel sections %s",
           darktable.num_openmp_threads, (int)dt_get_num_procs(), omp_get_dynamic() ? "(dynamic)" : "(static)");
+#endif
         k++;
         argv[k-1] = NULL;
         argv[k] = NULL;
