--- a/third_party/BUILD
+++ b/third_party/BUILD
@@ -36,6 +36,7 @@ filegroup(
         "//third_party/py/six:srcs",
         "//third_party/remoteapis:srcs",
         "//third_party/upb:srcs",
+        "//third_party/termux-patches:srcs",
         "//third_party/zlib:srcs",
     ],
 )
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -57,6 +57,7 @@ single_version_override(
     patch_strip = 1,
     patches = [
         "//third_party/grpc:00_disable_layering_check.patch",
+        "//third_party/termux-patches:grpc.patch",
     ],
 )
 
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -59,6 +59,23 @@
 bazel_dep(name = "rules_go", version = "0.39.1")
 bazel_dep(name = "upb", version = "0.0.0-20220923-a547704")
 
+single_version_override(
+    module_name = "c-ares",
+    patch_strip = 1,
+    patches = [
+       "//third_party/termux-patches:cares.patch",
+       "//third_party/termux-patches:cares-config.patch",
+    ],
+)
+
+single_version_override(
+    module_name = "protobuf",
+    patch_strip = 1,
+    patches = [
+       "//third_party/termux-patches:protobuf.patch",
+    ],
+)
+
 # =========================================
 # Java dependencies
 # =========================================

--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -73,6 +73,7 @@ single_version_override(
     patch_strip = 1,
     patches = [
         "//third_party/upb:01_remove_werror.patch",
+        "//third_party/termux-patches:upb-config.patch",
     ],
 )
 
