From 28c5766b1f4c8d91835161f61bc45e085f9c5702 Mon Sep 17 00:00:00 2001
From: anon <anon@localhost>
Date: Sun, 25 Sep 2022 04:28:30 +0800
Subject: [PATCH 1/2] mkfile

---
 Makefile | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/Makefile b/Makefile
index 5694e47..430f76a 100644
--- a/Makefile
+++ b/Makefile
@@ -22,16 +22,7 @@ sign:
 	gpg --detach-sign --yes "release/$(BIN)-$(VERSION).tar.gz"
 	rm "release/$(BIN)-$(VERSION).tar.gz"
 
-.PHONY: docs
-docs:
-	marked-man -i README.md -o "$(BIN).7"
-	gzip "$(BIN).7"
-
 .PHONY: install
 install:
 	install -Dm755 -t "$(BIN_DIR)/" checkrebuild
-	install -Dm644 -t "$(SHARE_DIR)/licenses/$(BIN)" LICENSE
 	install -Dm644 -t "$(SHARE_DIR)/libalpm/hooks" "$(BIN).hook"
-	install -Dm644 -t "$(SHARE_DIR)/man/man7" "$(BIN).7.gz"
-	install -dm755 "$(SHARE_DIR)/man/man1"
-	ln -Tsf "/usr/share/man/man7/$(BIN).7.gz" "$(SHARE_DIR)/man/man1/checkrebuild.1.gz"
-- 
2.37.3

