--- a/third_party/blink/renderer/core/layout/grid/grid_layout_algorithm.cc
+++ b/third_party/blink/renderer/core/layout/grid/grid_layout_algorithm.cc
@@ -691,8 +691,9 @@
 
   if (const auto* layout_subtree =
           GetConstraintSpace().GetGridLayoutSubtree()) {
-    auto& [grid_items, layout_data, subtree_size] =
-        sizing_tree.CreateSizingData();
+    auto& sizing_data = sizing_tree.CreateSizingData();
+    auto& grid_items = sizing_data.grid_items;
+    auto& layout_data = sizing_data.layout_data;
 
     const auto& node = Node();
     grid_items =
@@ -730,8 +731,9 @@
   const auto& track_collection = (track_direction == kForColumns)
                                      ? layout_data.Columns()
                                      : layout_data.Rows();
-  const auto& [begin_set_index, end_set_index] =
-      grid_item.SetIndices(track_direction);
+  const auto& set_indices = grid_item.SetIndices(track_direction);
+  const auto& begin_set_index = set_indices.begin;
+  const auto& end_set_index = set_indices.end;
 
   return (grid_item.BaselineGroup(track_direction) == BaselineGroup::kMajor)
              ? track_collection.MajorBaseline(begin_set_index)
@@ -1043,8 +1045,9 @@
   const auto& track_collection = (track_direction == kForColumns)
                                      ? subgridded_item.Columns(writing_mode)
                                      : subgridded_item.Rows(writing_mode);
-  const auto& [begin_set_index, end_set_index] =
-      subgridded_item->SetIndices(track_collection.Direction());
+  const auto& set_indices = subgridded_item->SetIndices(track_collection.Direction());
+  const auto& begin_set_index = set_indices.begin;
+  const auto& end_set_index = set_indices.end;
 
   const LayoutUnit extra_margin =
       (subgridded_item->BaselineGroup(track_direction) == BaselineGroup::kMajor)
@@ -1236,8 +1239,9 @@
                                      : subgridded_item.Rows(writing_mode);
 
   const auto margins = ComputeMarginsFor(space, item_style, constraint_space);
-  const auto& [begin_set_index, end_set_index] =
-      subgridded_item->SetIndices(track_collection.Direction());
+  const auto& set_indices = subgridded_item->SetIndices(track_collection.Direction());
+  const auto& begin_set_index = set_indices.begin;
+  const auto& end_set_index = set_indices.end;
 
   const auto margin_sum =
       (is_for_columns ? margins.InlineSum() : margins.BlockSum()) +
@@ -1656,8 +1660,10 @@
     //  in first/last baseline alignment within its start-most/end-most shared
     //  alignment context along that axis"
     // https://www.w3.org/TR/css-align-3/#baseline-sharing-group
-    const auto& [begin_set_index, end_set_index] =
-        grid_item.SetIndices(track_direction);
+    const auto& set_indices = grid_item.SetIndices(track_direction);
+    const auto& begin_set_index = set_indices.begin;
+    const auto& end_set_index = set_indices.end;
+        
     if (grid_item.BaselineGroup(track_direction) == BaselineGroup::kMajor) {
       track_collection.SetMajorBaseline(begin_set_index, baseline);
     } else {
@@ -1917,8 +1923,9 @@
     bool* opt_needs_additional_pass) const {
   DCHECK(sizing_subtree);
 
-  auto& [grid_items, layout_data, subtree_size] =
-      sizing_subtree.SubtreeRootData();
+  auto& sizing_data = sizing_subtree.SubtreeRootData();
+  auto& grid_items = sizing_data.grid_items;
+  auto& layout_data = sizing_data.layout_data;
 
   const bool is_for_columns = track_direction == kForColumns;
   const bool has_non_definite_track =
@@ -2062,8 +2069,10 @@
     const GridSizingSubtree& sizing_subtree,
     const CallbackFunc& callback_func,
     bool should_compute_min_max_sizes) const {
-  auto& [grid_items, layout_data, subtree_size] =
-      sizing_subtree.SubtreeRootData();
+  auto& sizing_data = sizing_subtree.SubtreeRootData();
+  auto& grid_items = sizing_data.grid_items;
+  auto& layout_data = sizing_data.layout_data;
+  auto& subtree_size = sizing_data.subtree_size;
 
   // If we know this subtree doesn't have nested subgrids we can exit early
   // instead of iterating over every grid item looking for them.
@@ -3527,7 +3536,9 @@
   DCHECK(out_row_break_between);
 
   const auto& container_space = GetConstraintSpace();
-  const auto& [grid_items, layout_data, tree_size] = sizing_tree.TreeRootData();
+  const auto& sizing_data = sizing_tree.TreeRootData();
+  const auto& grid_items = sizing_data.grid_items;
+  const auto& layout_data = sizing_data.layout_data;
 
   const auto* cached_layout_subtree = container_space.GetGridLayoutSubtree();
   const auto container_writing_direction =
@@ -3691,7 +3702,9 @@
 
   // TODO(ikilpatrick): Update |SetHasSeenAllChildren| and early exit if true.
   const auto& constraint_space = GetConstraintSpace();
-  const auto& [grid_items, layout_data, tree_size] = sizing_tree.TreeRootData();
+  const auto& sizing_data = sizing_tree.TreeRootData();
+  const auto& grid_items = sizing_data.grid_items;
+  const auto& layout_data = sizing_data.layout_data;
 
   const auto* cached_layout_subtree = constraint_space.GetGridLayoutSubtree();
   const auto container_writing_direction =
@@ -4395,8 +4408,9 @@
   DCHECK(!grid_item.IsOutOfFlow());
   DCHECK(!grid_item.is_subgridded_to_parent_grid);
 
-  const auto& [begin_set_index, end_set_index] =
-      grid_item.SetIndices(track_collection.Direction());
+  const auto& set_indices = grid_item.SetIndices(track_collection.Direction());
+  const auto& begin_set_index = set_indices.begin;
+  const auto& end_set_index = set_indices.end;
 
   if (start_offset) {
     *start_offset = track_collection.GetSetOffset(begin_set_index);
