--- a/configure.ac	2021-08-26 21:24:30.000000000 +0200
+++ b/configure.ac	2025-12-19 18:50:49.136621455 +0100
@@ -6,7 +6,7 @@
 AC_CONFIG_SRCDIR([src/enigma.cc])
 AC_CONFIG_MACRO_DIR([m4])
 
-AM_INIT_AUTOMAKE
+AM_INIT_AUTOMAKE([foreign])
 AC_CONFIG_HEADERS([src/config.h])
 AC_GNU_SOURCE
 
@@ -115,32 +115,12 @@
 CXXFLAGS_NOSDL="$CXXFLAGS"
 LIBS_NOSDL="$LIBS"
 
-AM_PATH_SDL2(2.0.0)
-AC_SUBST(SDL_LIBS)
-AC_SUBST(SDL_CFLAGS)
-dnl The flags are now substituted inside the Makefile, but not used for the
-dnl rest of the script. So we manually put them in. This is necessary if
-dnl SDL has not been installed in the canonical locations.
-
-CFLAGS="$CFLAGS $SDL_CFLAGS"
-CXXFLAGS="$CXXFLAGS $SDL_CFLAGS"
-LIBS="$LIBS $SDL_LIBS"
-
-dnl ----------------------------------------
-dnl Check for SDL support libraries
-dnl ----------------------------------------
-AC_CHECK_LIB(SDL2_image,main,,[AC_MSG_ERROR([SDL2_image is required to compile Enigma])])
-SDL_LIBS="$SDL_LIBS -lSDL2_image"
-AC_CHECK_LIB(SDL2_mixer,main,,[AC_MSG_ERROR([SDL2_mixer is required to compile Enigma])])
-SDL_LIBS="$SDL_LIBS -lSDL2_mixer"
-AC_CHECK_LIB(SDL2_ttf, main,,[AC_MSG_ERROR([SDL2_ttf is required to compile Enigma])])
-SDL_LIBS="$SDL_LIBS -lSDL2_ttf"
-
-if test "x$MACOSX" = xyes; then
- SDL2_PREFIX=`$SDL2_CONFIG $sdl2_config_args --variable=libdir`
- SDL_LIBS="-Wl,-rpath,@loader_path/../lib -framework AudioToolbox -framework AudioUnit -framework Cocoa -framework CoreAudio -framework IOKit -framework CoreFoundation -framework Carbon -framework CoreServices -framework ApplicationServices -framework Foundation -framework AppKit -framework OpenGL -lSDL2 -lSDL2_mixer -lSDL2_image -lSDL2_ttf $SDL2_PREFIX/libSDL2main.a"
-AC_SUBST(SDL2_PREFIX)
-fi
+PKG_CHECK_MODULES([SDL], [sdl2 >= 2.0.0 SDL2_image SDL2_mixer SDL2_ttf])
+AC_SUBST([SDL_CFLAGS])
+AC_SUBST([SDL_LIBS])
+CFLAGS="$CFLAGS $SDL_CFLAGS"
+CXXFLAGS="$CXXFLAGS $SDL_CFLAGS"
+LIBS="$LIBS $SDL_LIBS"
 
 dnl ---------------------------------------
 dnl Restoring variables to _NOSDL State
--- a/lib-src/enet/configure.ac	2021-08-26 21:24:30.000000000 +0200
+++ b/lib-src/enet/configure.ac	2025-12-19 18:50:49.128621384 +0100
@@ -1,6 +1,6 @@
 AC_INIT([libenet], [1.0])
 AC_CONFIG_SRCDIR([callbacks.c])
-AM_INIT_AUTOMAKE
+AM_INIT_AUTOMAKE([foreign])
 
 AC_PROG_CC
 AC_PROG_RANLIB
--- a/lib-src/zipios++/configure.ac	2021-08-26 21:24:30.000000000 +0200
+++ b/lib-src/zipios++/configure.ac	2025-12-19 18:50:49.128621384 +0100
@@ -1,7 +1,7 @@
 dnl Process this file with autoconf to produce a configure script.
 AC_INIT([zipios++], [0.1.5])
 AC_CONFIG_SRCDIR([src/backbuffer.h])
-AM_INIT_AUTOMAKE
+AM_INIT_AUTOMAKE([foreign])
 
 AC_CONFIG_HEADERS([zipios++/zipios-config.h])
 
--- a/src/main.cc	2021-08-26 21:24:30.000000000 +0200
+++ b/src/main.cc	2025-12-19 18:50:49.136621455 +0100
@@ -316,7 +316,7 @@
         if ( newfp == NULL ) {  // This happens on NT
             newfp = fopen((userStdPath + "/Output.log").c_str(), "w");
             if (newfp) {  // in case stdout is a macro
-                *stdout = *newfp;
+                // *stdout = *newfp;
             }
         }
         setvbuf(stdout, NULL, _IOLBF, BUFSIZ);   // Line buffered
@@ -325,7 +325,7 @@
         if ( newfp == NULL ) {  // This happens on NT
             newfp = fopen((userStdPath + "/Error.log").c_str(), "w");
             if (newfp) {  // in case stderr is a macro
-                *stderr = *newfp;
+                // *stderr = *newfp;
             }
         }
         setbuf(stderr, NULL);   // No buffering
--- a/src/nls.cc	2021-08-26 21:24:30.000000000 +0200
+++ b/src/nls.cc	2025-12-19 18:50:49.136621455 +0100
@@ -93,7 +93,7 @@
 
 std::unique_ptr<std::istream> TinyGetTextFileSystem::open_file(const std::string& filename) {
     std::basic_ifstream<char> ifs(filename, ios::binary | ios::in);
-    unique_ptr<std::basic_ifstream<char>> isptr = make_unique<std::basic_ifstream<char>>(move(ifs));
+    unique_ptr<std::basic_ifstream<char>> isptr = make_unique<std::basic_ifstream<char>>(std::move(ifs));
     return isptr;
 }
