--- a/setup.py
+++ b/setup.py
@@ -771,7 +771,7 @@
         # _queue module
         self.add(Extension("_queue", ["_queuemodule.c"]))
         # _statistics module
-        self.add(Extension("_statistics", ["_statisticsmodule.c"]))
+        self.add(Extension("_statistics", ["_statisticsmodule.c"], libraries=["m"]))
 
         # Modules with some UNIX dependencies -- on by default:
         # (If you have a really backward UNIX, select and socket may not be
