--- a/third_party/dawn/src/dawn/native/XlibXcbFunctions.cpp
+++ b/third_party/dawn/src/dawn/native/XlibXcbFunctions.cpp
@@ -17,7 +17,11 @@
 namespace dawn::native {
 
     XlibXcbFunctions::XlibXcbFunctions() {
+#ifndef __TERMUX__
         if (!mLib.Open("libX11-xcb.so.1") ||
+#else
+        if (!mLib.Open("libX11-xcb.so") ||
+#endif
             !mLib.GetProc(&xGetXCBConnection, "XGetXCBConnection")) {
             mLib.Close();
         }
