--- blender/intern/cycles/kernel/CMakeLists.txt	2023-12-25 17:41:40.320821135 +0530
+++ blender.mod/intern/cycles/kernel/CMakeLists.txt	2023-12-25 17:50:01.795511212 +0530
@@ -1132,17 +1132,6 @@
   set_source_files_properties(device/cpu/kernel_avx2.cpp PROPERTIES COMPILE_FLAGS "${CYCLES_AVX2_KERNEL_FLAGS}")
 endif()
 
-# Warnings to avoid using doubles in the kernel.
-if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_C_COMPILER_ID MATCHES "Clang")
-  add_check_cxx_compiler_flags(
-    CMAKE_CXX_FLAGS
-    _has_cxxflag_float_conversion "-Werror=float-conversion"
-    _has_cxxflag_double_promotion "-Werror=double-promotion"
-  )
-  unset(_has_cxxflag_float_conversion)
-  unset(_has_cxxflag_double_promotion)
-endif()
-
 cycles_add_library(cycles_kernel "${LIB}"
   ${SRC_KERNEL_DEVICE_CPU}
   ${SRC_KERNEL_DEVICE_CUDA}

