--- a/build/install-build-deps.sh
+++ b/build/install-build-deps.sh
@@ -395,6 +395,7 @@
   msttcorefonts
   python-dev
   python-setuptools
+  snapcraft
   ttf-dejavu-core
   ttf-indic-fonts
   ttf-kochi-gothic
@@ -580,9 +581,6 @@
 if package_exists libinput-dev; then
     dev_list="${dev_list} libinput-dev"
 fi
-if package_exists snapcraft; then
-    dev_list="${dev_list} snapcraft"
-fi
 
 # Cross-toolchain strip is needed for building the sysroots.
 if package_exists binutils-arm-linux-gnueabihf; then
