--- a/test/common/index.js
+++ b/test/common/index.js
@@ -129,19 +129,7 @@
 const isLinux = process.platform === 'linux';
 const isOSX = process.platform === 'darwin';
 const isASan = process.config.variables.asan === 1;
-const isPi = (() => {
-  try {
-    // Normal Raspberry Pi detection is to find the `Raspberry Pi` string in
-    // the contents of `/sys/firmware/devicetree/base/model` but that doesn't
-    // work inside a container. Match the chipset model number instead.
-    const cpuinfo = fs.readFileSync('/proc/cpuinfo', { encoding: 'utf8' });
-    const ok = /^Hardware\s*:\s*(.*)$/im.exec(cpuinfo)?.[1] === 'BCM2835';
-    /^/.test('');  // Clear RegExp.$_, some tests expect it to be empty.
-    return ok;
-  } catch {
-    return false;
-  }
-})();
+const isPi = false;
 
 const isDumbTerminal = process.env.TERM === 'dumb';
 
