--- a/components/performance_manager/resource_attribution/context_collection.h
+++ b/components/performance_manager/resource_attribution/context_collection.h
@@ -28,7 +28,7 @@
   ContextCollection(const ContextCollection& other);
   ContextCollection& operator=(const ContextCollection& other);
 
-  friend constexpr bool operator==(const ContextCollection&,
+  friend bool operator==(const ContextCollection&,
                                    const ContextCollection&) = default;
 
   // Adds `context` to the collection.
