diff -uNr a/runtime/libpgmath/tools/mth_mask.awk b/runtime/libpgmath/tools/mth_mask.awk
--- a/runtime/libpgmath/tools/mth_mask.awk
+++ b/runtime/libpgmath/tools/mth_mask.awk
@@ -18,11 +18,11 @@
  *\n\
  */\n\
 \n\n\
+#include \"mth_intrinsics.h\" \n\
+#include \"mth_tbldefs.h\" \n\
 #ifdef __cplusplus\n\
 extern \"C\" {\n\
 #endif\n\n\
-#include \"mth_intrinsics.h\" \n\
-#include \"mth_tbldefs.h\" \n\
 \n\n\
 static const vrs4_t Csp1_4={1.0, 1.0, 1.0, 1.0}; \n\
 static const vrd2_t Cdp1_2={1.0, 1.0}; \n\
