--- a/bin/install_megadrivers.py
+++ b/bin/install_megadrivers.py
@@ -24,6 +24,7 @@

 import argparse
 import os
+import shutil
 
 def main():
     parser = argparse.ArgumentParser()
@@ -55,7 +54,7 @@
         if os.path.lexists(abs_driver):
             os.unlink(abs_driver)
         print('installing {} to {}'.format(args.megadriver, abs_driver))
-        os.link(master, abs_driver)
+        shutil.copyfile(master, abs_driver)
 
         try:
             ret = os.getcwd()
