diff --git a/app/src/ui/index.tsx b/app/src/ui/index.tsx
--- a/app/src/ui/index.tsx
+++ b/app/src/ui/index.tsx
@@ -89,7 +89,7 @@
 
 // Tell dugite where to find the git environment,
 // see https://github.com/desktop/dugite/pull/85
-process.env['LOCAL_GIT_DIRECTORY'] = Path.resolve(__dirname, 'git')
+process.env['LOCAL_GIT_DIRECTORY'] ??= Path.resolve(__dirname, 'git')
 
 // Ensure that dugite infers the GIT_EXEC_PATH
 // based on the LOCAL_GIT_DIRECTORY env variable
