diff --git a/build_files/cmake/platform/platform_unix.cmake b/build_files/cmake/platform/platform_unix.cmake
index 244641e..3b39c7f 100644
--- a/build_files/cmake/platform/platform_unix.cmake
+++ b/build_files/cmake/platform/platform_unix.cmake
@@ -397,7 +397,6 @@ if(WITH_BOOST)
     if(WITH_USD AND USD_PYTHON_SUPPORT)
       list(APPEND __boost_packages python${PYTHON_VERSION_NO_DOTS})
     endif()
-    list(APPEND __boost_packages system)
     set(Boost_NO_WARN_NEW_VERSIONS ON)
     find_package(Boost 1.48 COMPONENTS ${__boost_packages})
     if(NOT Boost_FOUND)
