--- blender/intern/guardedalloc/intern/mallocn_intern.h	2023-08-03 07:40:38.037447528 +0530
+++ blender.mod/intern/guardedalloc/intern/mallocn_intern.h	2023-08-03 19:00:54.884799067 +0530
@@ -23,7 +23,7 @@
 
 #if defined(HAVE_MALLOC_STATS_H)
 #  include <malloc.h>
-#  define HAVE_MALLOC_STATS
+#  define _HAVE_MALLOC_STATS
 #elif defined(__FreeBSD__)
 #  include <malloc_np.h>
 #elif defined(__NetBSD__) || defined(__OpenBSD__)
